docs-src/references/dynamic-configuration.md

```diff
@@ -0,0 +1,214 @@
+Also, ensure that you test your changes before setting these in production.
+
+## Format
+
+To override the default dynamic configuration values, specify your custom values and constraints for the dynamic configuration keys that you want to change in a YAML configuration file.
+Use the following format when creating your dynamic configuration file.
+
+```yaml
+testGetBoolPropertyKey:
+  - value: false
+  - value: true
+    constraints:
+      namespace: "your-namespace"
+  - value: false
+    constraints:
+      namespace: "your-other-namespace"
+testGetDurationPropertyKey:
+  - value: "1m"
+    constraints:
+      namespace: "your-namespace"
+      taskQueueName: "longIdleTimeTaskqueue"
+testGetFloat64PropertyKey:
+  - value: 12.0
+    constraints:
+      namespace: "your-namespace"
+testGetMapPropertyKey:
+  - value:
+      key1: 1
+      key2: "value 2"
+      key3:
+        - false
+        - key4: true
+          key5: 2.0
+```
+
+### Constraints
+
+Some dynamic configuration keys are set globally, and others can be customized on a Namespace or Task Queue level.
+
+- To set global values for the configuration key with no constraints, use:
+
+  ```yaml
+  keyName:
+    - value: "my-value"
+  ```
+
+- For keys that can be customized at Namespace level, you can specify multiple values for different Namespaces in addition to one default value that applies globally to all Namespaces.
+  To set values at a Namespace level, use `namespace` (String) as shown in the following example.
+
+  ```yaml
+  keyName:
+    - constraints: {} # Sets default value that applies to all Namespaces
+      value: "value-for-all-the-rest"
+  - constraints: {namespace: "namespace1"} # Sets value for "namespace1" Namespace
+      value: "value-for-namespace1"
+    - constraints: {namespace: "namespace2"}
+      value: "value-for-namespace2"
+  ```
+
+- For keys that can be customized at a Task Queue level, you can specify Task Queue name and Task type in addition to Namespace.
+  To set values at a Task Queue level, use `taskqueueName` (String) with `taskType` (optional; supported values: `Workflow` and `Activity`), as shown in the following exmaple.
+
+  ```yaml
+  keyName:
+  - constraints: {namespace: "namespace1", taskQueueName: "tq"}  #Applies to both Workflow and Activity tasks on the "tq" Task Queue.
+    value: 44
+  - constraints: {namespace: "namespace1", taskQueueName: "other-tq", taskType: "Activity"} #Applies to Activity tasks on the "other_tq" Task Queue.
+    value: 33
+  - constraints: {namespace: "namespace2"}  #Applies to all task queues in "namespace2".
+    value: 22
+  - constraints: {}  # Applies to all other task queues in "namespace1" and all other Namespaces.
+    value: 11
+  ```
+
+Note that these values are not applied first to last; values set on specific constraints override default values defined for that configuration key.
```

Comment:
I don't think this makes sense. The point is: for a specific key, "more constrained" values take priority over "less constrained", and also of course over the default, regardless of the order in the yaml file.

As written, it just says constrained values override the default value, which doesn't say anything about the priority among multiple constrained values.

Reply:
yeah; it was a little wonky. Have updated to 
`Note that the values set with most constraints take priority over values that are set with fewer constraints, regardless of the order in which they are set in the dynamic configuration key.` > is this clearer?